src/npc.h

```diff
@@ -72,7 +72,7 @@ class NpcEventsHandler
 
 private:
 	Npc* npc;
-	NpcScriptInterface* scriptInterface;
+	std::shared_ptr<NpcScriptInterface> scriptInterface;
```

Comment:
How about making it a `unique_ptr` to avoid the extra cost? Since the Npc class just needs to access it, you can store a reference to the unique_ptr there.